File: functions/utils/respond.js
const errors = require('./errors');

module.exports = {
  success(data) {
    return {
      statusCode: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: JSON.stringify(data),
    };
  },
  created(data) {

  },
  badRequest(e) {
    return {
      statusCode: 500,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: JSON.stringify(e.message)
    };
  },
  error(e) {
    if (e instanceof errors.BadRequest) {
      return this.badRequest(e)
    } else {
      return {
        statusCode: 500,
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Credentials': true,
        },
        body: JSON.stringify(e.message)
      };
    }
  },
  notFound(data) {

  }
};

File: nlp/utils/respond.py
import json

def success(data):
  return {
    'statusCode': 200,
    'headers': {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Credentials': 'true',
    },
    'body': json.dumps(data),
  }

def bad_request(e):
  return {
    'statusCode': 400,
    'headers': {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Credentials': 'true',
    },
    'body': json.dumps(e)
  }

def error(e):
  return {
    'statusCode': 500,
    'headers': {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Credentials': 'true',
    },
    'body': json.dumps(e)
  }
